### PR TITLE
Factorio: update API use

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
updates various Factorio API use to current AP standards.

## How was this tested?
unittests and a single local gen

### CODE REVIEW

**Comment by NewSoupVi** [comment]
Thank you 🎉

File: worlds/factorio/Mod.py [hunk 1]
```diff
@@ -129,40 +130,40 @@ def flop_random(low, high, base=None):
         "base_tech_table": base_tech_table,
         "tech_to_progressive_lookup": tech_to_progressive_lookup,
         "mod_name": mod_name,
-        "allowed_science_packs": multiworld.max_science_pack[player].get_allowed_packs(),
-        "custom_technologies": multiworld.worlds[player].custom_technologies,
+        "allowed_science_packs": world.options.max_science_pack.get_allowed_packs(),
+        "custom_technologies": world.custom_technologies,
         "tech_tree_layout_prerequisites": world.tech_tree_layout_prerequisites,
         "slot_name": multiworld.player_name[player], "seed_name": multiworld.seed_name,
```
Comment (Exempt-Medic) [inline worlds/factorio/Mod.py:136]: Could do this
```suggestion
        "slot_name": world.player_name, "seed_name": multiworld.seed_name,
```

File: worlds/factorio/Mod.py [hunk 2]
```diff
@@ -129,40 +130,40 @@ def flop_random(low, high, base=None):
...
+    if world.options.satellite == Options.Satellite.option_randomize_recipe:
         template_data["free_sample_blacklist"]["satellite"] = 1
 
-    template_data["free_sample_blacklist"].update({item: 1 for item in multiworld.free_sample_blacklist[player].value})
-    template_data["free_sample_blacklist"].update({item: 0 for item in multiworld.free_sample_whitelist[player].value})
+    template_data["free_sample_blacklist"].update({item: 1 for item in world.options.free_sample_blacklist.value})
+    template_data["free_sample_blacklist"].update({item: 0 for item in world.options.free_sample_whitelist.value})
 
     zf_path = os.path.join(output_directory, versioned_mod_name + ".zip")
     mod = FactorioModFile(zf_path, player=player, player_name=multiworld.player_name[player])
```
Comment (Exempt-Medic) [inline worlds/factorio/Mod.py:169]:
```suggestion
    mod = FactorioModFile(zf_path, player=player, player_name=world.player_name)
```

File: worlds/factorio/__init__.py [hunk 3]
```diff
@@ -436,66 +434,66 @@ def make_balanced_recipe(self, original: Recipe, pool: typing.Set[str], factor:
...
         new_recipe = self.make_quick_recipe(
             Recipe(bridge, "crafting", {"replace_1": 1, "replace_2": 1, "replace_3": 1,
                                         "replace_4": 1, "replace_5": 1, "replace_6": 1},
                    {bridge: 1}, 10),
-            sorted(science_pack_pools[self.multiworld.max_science_pack[self.player].get_ordered_science_packs()[0]]),
-            ingredients_offset=ingredients_offset)
+            sorted(science_pack_pools[self.options.max_science_pack.get_ordered_science_packs()[0]]),
+            ingredients_offset=ingredients_offset.value)
         for ingredient_name in new_recipe.ingredients:
             new_recipe.ingredients[ingredient_name] = self.multiworld.random.randint(50, 500)
```
Comment (Exempt-Medic) [inline worlds/factorio/__init__.py:490]: If you're changing the rest, might as well do this one
```suggestion
            new_recipe.ingredients[ingredient_name] = self.random.randint(50, 500)
```

File: worlds/factorio/__init__.py [hunk 4]
```diff
@@ -545,7 +543,8 @@ def __init__(self, player: int, name: str, address: int, parent: Region):
 
         self.ingredients = {Factorio.ordered_science_packs[self.complexity]: 1}
         for complexity in range(self.complexity):
-            if parent.multiworld.tech_cost_mix[self.player] > parent.multiworld.random.randint(0, 99):
+            if (parent.multiworld.worlds[self.player].options.tech_cost_mix >
+                    parent.multiworld.random.randint(0, 99)):
```
Comment (Exempt-Medic) [inline worlds/factorio/__init__.py:547]: Could do this too, I suppose
```suggestion
                    parent.multiworld.worlds[self.player].random.randint(0, 99)):
```

**Review by Exempt-Medic — APPROVED** [review]
The changes LGTM. Generations are completely different, presumably because of the random changes, but the playthrough seems similar (I don't know the game really at all though, just going off of item names). A few minor comments.

**Review by Exempt-Medic — APPROVED** [review]
LGTM. Random generations didn't hit any old options stuff and I didn't find any with a search. Generations before and after are consiste if the `random` changes are reverted. Comments were fully addressed

**Review by qwint — APPROVED** [review]
pure code review; did not run gens/tests/etc., did not check for completeness

updates all looks fine, got a little worried about changing the 'world' parameter in technology.get_custom() to the World instance instead of MultiWorld, but digging through that codepath it seems to be fine/correct

**Review by silasary — APPROVED** [review]

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**